code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/EnvironmentCreateValidationFilter.java

```diff
@@ -0,0 +1,32 @@
+package io.cattle.platform.servicediscovery.api.filter;
+
+import static io.cattle.platform.core.model.tables.EnvironmentTable.ENVIRONMENT;
+import io.cattle.platform.core.model.Environment;
+import io.cattle.platform.iaas.api.filter.common.AbstractDefaultResourceManagerFilter;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.ResourceManager;
+import io.github.ibuildthecloud.gdapi.validation.ValidationErrorCodes;
+
+import javax.inject.Inject;
+
+public class EnvironmentCreateValidationFilter extends AbstractDefaultResourceManagerFilter {
+    @Inject
+    ObjectManager objectManager;
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { Environment.class };
+    }
+
+    @Override
+    public Object create(String type, ApiRequest request, ResourceManager next) {
+        Environment env = request.proxyRequestObject(Environment.class);
+
+        if (env.getName().startsWith("-") || env.getName().endsWith("-")) {
```

Comment:
Hmm.  Something I never checked before, but I'm guessing that the name has already been trimmed of leading and trailing white space?